src/main/java/in/nivethitha/model/CouponDetails.java

```diff
@@ -0,0 +1,73 @@
+package in.nivethitha.model;
+
+import java.util.Date;
+
+public class CouponDetails {
+	private int id;
+	private String shoppingSiteName;
+	private int discount;
+	private double purchaseAmount;
+	private Date startDate;

```

Comment:
Use LocalDate instead of date. LocalDate has more utility methods like isAfter, isBefore etc..